tests/grpc/grpc_test.go

```diff
@@ -227,3 +228,137 @@ func Test_Grpc_Terminate_Recursive(t *testing.T) {
 		})
 	}
 }
+
+func Test_Grpc_ReuseInstanceIDSkip(t *testing.T) {
+	delayTime := 2 * time.Second
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("SingleActivity", func(ctx *task.OrchestrationContext) (any, error) {
+		var input string
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		var output string
+		ctx.CreateTimer(delayTime).Await(nil)
+		err := ctx.CallActivity("SayHello", task.WithActivityInput(input)).Await(&output)
+		return output, err
+	})
+	r.AddActivityN("SayHello", func(ctx task.ActivityContext) (any, error) {
+		var name string
+		if err := ctx.GetInput(&name); err != nil {
+			return nil, err
+		}
+		return fmt.Sprintf("Hello, %s!", name), nil
+	})
+
+	cancelListener := startGrpcListener(t, r)
+	defer cancelListener()
+	instanceIDs := api.InstanceID("SKIP_IF_RUNNING_OR_COMPLETED")
+	ReuseIdOption := api.OrchestrationIDReuseOption{
+		CreateOrchestrationAction: protos.CreateOrchestrationAction_SKIP,
+		OrchestrationStatuses: []protos.OrchestrationStatus{
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_RUNNING,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_COMPLETED,
+			protos.OrchestrationStatus_ORCHESTRATION_STATUS_PENDING,
+		},
+	}
+
+	id, err := grpcClient.ScheduleNewOrchestration(ctx, "SingleActivity", api.WithInput("世界"), api.WithInstanceID(instanceIDs))
+	require.NoError(t, err)
+	// random sleep a time from 0 to 5 seconds, to allow previous orchestration instance to set in different status
+	randomeDuration := time.Duration(rand.Intn(6)) * time.Second
+	time.Sleep(randomeDuration)
```

Comment:
Is the intent just to wait for the orchestration to get into a running state? If so, instead of sleeping, you should do `grpcClient.WaitForOrchestrationStart(...)`. Random sleeps is probably not a great idea because it could cause the test to become flakey.